### PR TITLE
Tweak survey infrastructure

### PR DESCRIPTION
**Pull request by subsymbolic**
Task/Issue URL: https://app.asana.com/0/361428290920652/1198222027640374

**Description**:
Some small tweaks to our survey infrastructure
* only show surveys without a set day to installers who have been with us for more than 30 days
* restrict surveys to US locale
* fix issue where CTA could still show after survey completed
* small UI tweaks

**Steps to test this PR**:

**_Setup_**
1. Open SurveyService and update to our sandbox endpoint "http://ddg-sandbox.s3.amazonaws.com/survey/survey-mobile.json"

**_Test restrictions_**
1. Run a fresh install on a non-US locale and ensure that the survey CTA does not show on the home screen (minimum installation days needs to be 30 and locale must be US).
1. Open CTAViewModel and change line 248 to be `val daysInstalled = 30L`, run the app again and ensure that the survey does show on the home screen (locale needs to be US).
1. Change locale to US.
1. Run the app again, the survey should now show.

**_Test survey is dismissed when done_**
1. Run through the survey completing it.
1. Ensure that the CTA disappears from the home screen.

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: app/src/main/java/com/duckduckgo/app/survey/ui/SurveyActivity.kt [hunk 1]
```diff
@@ -153,7 +153,7 @@ class SurveyActivity : DuckDuckGoActivity() {
         }
 
         override fun shouldInterceptRequest(view: WebView, request: WebResourceRequest): WebResourceResponse? {
-            if (request.isForMainFrame && request.url.host == "www.duckduckgo.com") {
+            if (request.url.host == "duckduckgo.com") {
```
Comment (cmonfortep) [inline app/src/main/java/com/duckduckgo/app/survey/ui/SurveyActivity.kt:156]: questions: why are we removing the `isForMainFrame` check? I assume we know it's going to be always `true`?

File: app/src/androidTest/java/com/duckduckgo/app/survey/ui/SurveyViewModelTest.kt [hunk 2]
```diff
@@ -63,7 +64,7 @@ class SurveyViewModelTest {
     @Before
     fun before() {
         MockitoAnnotations.initMocks(this)
-        testee = SurveyViewModel(mockSurveyDao, mockStatisticsStore, mockAppInstallStore)
+        testee = SurveyViewModel(mockSurveyDao, mockStatisticsStore, mockAppInstallStore, coroutineTestRule.testDispatcherProvider)
```
Comment (cmonfortep) [inline app/src/androidTest/java/com/duckduckgo/app/survey/ui/SurveyViewModelTest.kt:67]: 👍

File: app/src/androidTest/java/com/duckduckgo/app/cta/ui/CtaViewModelTest.kt [hunk 3]
```diff
@@ -180,16 +180,43 @@ class CtaViewModelTest {
     }
 
     @Test
-    fun whenScheduledSurveyChangesAndInstalledDaysMatchThenCtaIsSurvey() {
-        val survey = Survey("abc", "http://example.com", 1, SCHEDULED)
+    fun whenScheduledSurveyChangesAndInstalledDaysMatchAndLocleIsUsThenCtaIsSurvey() = coroutineRule.runBlocking {
```
Comment (cmonfortep) [inline app/src/androidTest/java/com/duckduckgo/app/cta/ui/CtaViewModelTest.kt:183]: typo in `Locle` -> `Locale`

File: app/src/androidTest/java/com/duckduckgo/app/cta/ui/CtaViewModelTest.kt [hunk 4]
```diff
@@ -180,16 +180,43 @@ class CtaViewModelTest {
...
+        testee.onSurveyChanged(Survey("abc", "http://example.com", 1, SCHEDULED))
+        val value = testee.refreshCta(coroutineRule.testDispatcher, isBrowserShowing = false, site = null, locale = Locale.US)
+        assertTrue(value is HomePanelCta.Survey)
+    }
+
+    @Test
+    fun whenScheduledSurveyChangesAndInstalledDaysMatchButLocleIsNotUsThenCtaIsNotSurvey() = coroutineRule.runBlocking {
```
Comment (cmonfortep) [inline app/src/androidTest/java/com/duckduckgo/app/cta/ui/CtaViewModelTest.kt:190]: typo in testName

**Review by cmonfortep — APPROVED** [review]
LGTM. Tested and works as expected 👍 

Just some minor comments and one question.

File: app/src/main/java/com/duckduckgo/app/survey/ui/SurveyActivity.kt (hunk 1 above)
Comment (subsymbolic) [reply app/src/main/java/com/duckduckgo/app/survey/ui/SurveyActivity.kt:156]: No the opposite, it's false. Something in surveymonkey has changed and it's not loading our page as a mainframe request anymore so we need to drop that requirement. As soon as we start seeing requests for "duckduckgo.com" we know we have finished with the survey.

**Comment by subsymbolic** [comment]
Thanks for the review @cmonfortep!

**Pull request merged**